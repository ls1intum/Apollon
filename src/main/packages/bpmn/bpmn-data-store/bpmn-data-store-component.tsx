import React, { FunctionComponent } from 'react';
import { ThemedPath } from '../../../components/theme/themedComponents';
import { Multiline } from '../../../utils/svg/multiline';
import { BPMNDataStore } from './bpmn-data-store';

export const BPMNDataStoreComponent: FunctionComponent<Props> = ({ element, fillColor, strokeColor, textColor }) => (
  <g>
    <ThemedPath
      d={`M 0 10 L 0 ${element.bounds.height - 10} A ${element.bounds.width / 2} 10 0 0 0 ${element.bounds.width} ${
        element.bounds.height - 10
      } L ${element.bounds.width} 10 A ${element.bounds.width / 2} 10 180 0 0 0 10`}
      strokeColor={strokeColor || element.strokeColor}
      fillColor={fillColor || element.fillColor}
    />

    <ThemedPath
      d={`M 0 30 A ${element.bounds.width / 2} 10 0 0 0 ${element.bounds.width} 30`}
      strokeColor={strokeColor || element.strokeColor}
      fillColor="transparent"
    />
    <ThemedPath
      d={`M 0 20 A ${element.bounds.width / 2} 10 0 0 0 ${element.bounds.width} 20`}
      strokeColor={strokeColor || element.strokeColor}
      fillColor="transparent"
    />
    <ThemedPath
      d={`M 0 10 A ${element.bounds.width / 2} 10 0 0 0 ${element.bounds.width} 10`}
      strokeColor={strokeColor || element.strokeColor}
      fillColor="transparent"
    />

    <Multiline
      x={element.bounds.width / 2}
      y={element.bounds.height + 20}
      width={element.bounds.width * 2}
      fill={textColor || element.textColor}
      lineHeight={16}
      capHeight={11}
      verticalAnchor="start"
    >
      {element.name}
    </Multiline>
  </g>
);

interface Props {
  element: BPMNDataStore;
  fillColor?: string;
  strokeColor?: string;
  textColor?: string;
}
