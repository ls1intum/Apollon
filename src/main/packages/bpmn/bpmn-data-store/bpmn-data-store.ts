import { BPMNElementType } from '..';
import { ILayer } from '../../../services/layouter/layer';
import { ILayoutable } from '../../../services/layouter/layoutable';
import { UMLElementType } from '../../uml-element-type';
import { UMLElementFeatures } from '../../../services/uml-element/uml-element-features';
import { UMLContainer } from '../../../services/uml-container/uml-container';

export class BPMNDataStore extends UMLContainer {
  static features: UMLElementFeatures = { ...UMLContainer.features, resizable: false };

  type: UMLElementType = BPMNElementType.BPMNDataStore;

  render(canvas: ILayer): ILayoutable[] {
    return [this];
  }
}
