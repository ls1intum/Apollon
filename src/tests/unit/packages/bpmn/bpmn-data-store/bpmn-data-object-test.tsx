import { wrappedRender } from '../../../test-utils/render';
import * as React from 'react';
import { BPMNDataStoreComponent } from '../../../../../main/packages/bpmn/bpmn-data-store/bpmn-data-store-component';
import { BPMNDataStore } from '../../../../../main/packages/bpmn/bpmn-data-store/bpmn-data-store';

it('render the bpmn-data-store-component', () => {
  const dataStore: BPMNDataStore = new BPMNDataStore({ name: 'Data Store' });
  const { getByText, baseElement } = wrappedRender(
    <svg>
      <BPMNDataStoreComponent element={dataStore} />
    </svg>,
  );
  expect(getByText(dataStore.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
