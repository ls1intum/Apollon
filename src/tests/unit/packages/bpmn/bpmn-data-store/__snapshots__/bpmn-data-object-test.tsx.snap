// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the bpmn-data-store-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <path
          class="sc-gEvEer fMgKSV"
          d="M 0 10 L 0 90 A 100 10 0 0 0 200 90 L 200 10 A 100 10 180 0 0 0 10"
          fill="white"
          stroke="black"
        />
        <path
          class="sc-gEvEer JpXgg"
          d="M 0 30 A 100 10 0 0 0 200 30"
          fill="transparent"
          stroke="black"
        />
        <path
          class="sc-gEvEer JpXgg"
          d="M 0 20 A 100 10 0 0 0 200 20"
          fill="transparent"
          stroke="black"
        />
        <path
          class="sc-gEvEer JpXgg"
          d="M 0 10 A 100 10 0 0 0 200 10"
          fill="transparent"
          stroke="black"
        />
        <text
          pointer-events="none"
          text-anchor="middle"
          width="400"
          x="100"
          y="120"
        >
          <tspan
            dy="11"
            x="100"
          >
            Data Store
          </tspan>
        </text>
      </g>
    </svg>
  </div>
</body>
`;
